text/0022-response-context.md

```diff
@@ -0,0 +1,144 @@
+
+# Background
+
+This is necessary to implement the `Failed HTTP Client requests automatically result in Events` feature that is described in this [DACI](https://www.notion.so/sentry/Failed-HTTP-Client-requests-automatically-result-in-Events-f6c21d2a58ce4f2c889a823fd1da0044).
+
+Since the `Response` metadata is PII sensitive, it should be properly mapped, documented and scrubbed.
+
+The [Dart Dio HTTP Client](https://docs.sentry.io/platforms/dart/configuration/integrations/dio/) integration already adds the `Response` field in the `Contexts` interface.
+
+# Supporting Data
+
+See [Supporting data](https://www.notion.so/sentry/Failed-HTTP-Client-requests-automatically-result-in-Events-f6c21d2a58ce4f2c889a823fd1da0044#0ca951d5216742dbaab02f5fd33b8fb5) section in the [DACI](https://www.notion.so/sentry/Failed-HTTP-Client-requests-automatically-result-in-Events-f6c21d2a58ce4f2c889a823fd1da0044).
+
+# Options Considered
+
+## Option 1
+
+Adding a `Response` interface directly in the [Event payload](https://develop.sentry.dev/sdk/event-payloads/).
+
+```json
+{
+  "response": {
+    // ..
+  }
+}
+```
+
+## Option 2 (Chosen)
+
+Adding a `Response` interface in the [Contexts interface](https://develop.sentry.dev/sdk/event-payloads/contexts/).
+
+```json
+{
+    "contexts": {
+        "response": {
+            "type": "response"
+            // ...
+  }
+}
+```
+
+## Option 3
+
+Expand the `Request` interface adding the missing fields.
+
+Data scrubbing should consider response headers when scrubbing.
+
+If we do that, the `Request` docs should be ammended that it contains the `Response` data as well otherwise it's semantically wrong.
+
+# Proposal
+
+The proposal is the Option 2, adding a `Response` interface in the [Contexts interface](https://develop.sentry.dev/sdk/event-payloads/contexts/).
+
+By doing the Option 2, we can keep the `Request` interface as it is and we don't need to change the data scrubbing rules for the `Request` field.
+
+Adding it as part of the `Contexts`, we get a lot for free such as retained arbitrary fields and back compatibility.
+
+```json
+{
+  "contexts": {
+    "response": {
+      "type": "response",
+      "cookies": "PHPSESSID=298zf09hf012fh2; csrftoken=u32t4o3tb3gg43; _gat=1;",
+      "headers": {
+        "content-type": "text/html"
+      },
+      "inferred_content_type": "text/html",
+      "status_code": 500,
+      "is_redirect": false,
+      "body_size": 1000,
+      "arbitrary_field": "arbitrary" // arbitrary and retained fields for backwards compatibility when adding new fields
+    }
+  }
+}
+```
+
+* `type`: `response` as `String`.
+* `cookies`: Can be given unparsed as `String`, as `Dictionary`, or as a `List of Tuples`.
+* `headers`: A `Dictionary` of submitted headers, this requires a special treatment in the data scrubbing rules.
+* `status_code`: The HTTP status code, `Integer`.
+* `is_redirect`: A `Boolean` indicating if the response was a redirect.
```

Comment:
To clarify: this is `false` if the request was _redirected_?

Reply:
If this can be inferred directly from the status code, then we could get rid of this attribute. Otherwise, we should also have more flags like `is_success`, `is_server_error`, `is_client_error` -- or even an interpreted status field. I believe this is a bit redundant, however.

Reply:
makes sense, removing it.